### ISSUE TITLE
hardcore mode

### ISSUE THREAD

**Issue by seiyria** [IdleLands/IdleLands#262]
hard mode characters:

- [x] should have a flag saying they're hardcore
- [ ] link to original non-hardcore character
   - [ ] should use the player uuid as a key required to generate the character, else the server won't let it happen
   - [ ] should require that you've got a character already
   - [ ] should have a checkbox, hidden, on the signup page
- [ ] should probably get an achievement buff at the start (it should definitely boost their base HP by a lot) and some equipment specialized for it. all hardcore characters should start with the same items to prevent scumming. maybe start with some core personalities too. they should start with either a special pet, or a set of pets that let them focus however they want to, which only cost 1 gold via this achievement.  Some examples of achievements **for the Hardcore player**:
   - [x] "Hardcore" -- Play as a Hardcore character
   - [x] "Deceased" -- Currently be dead as a Hardcore character
   - [ ] "Revived" -- Have at least two deaths as a Hardcore character
   - [ ] "Stepmaster" -- Have X steps as your best as a Hardcore character
   - [ ] "Goldmaster" -- Have X gold as your best as a Hardcore character
   - [ ] "Deathless" -- Have X ascensions currently a Hardcore character
- [x] statistics for Hardcore characters' best performance should be updated on character death
- [ ] should not be able to do quests or roll premium [hide in ui]
   - [ ] any API calls that do quests or roll premium should return failure messages
   - [ ] they should get some base premium bonuses to buff duration, choice log size, etc
   - [ ] when clicking the Premium tab, should automatically redirect to the Gold route, as ILP is not available
- [x] should reset to level 1, ascension 0 when they die and have a "game over" note in their adventure log (maybe with a "reset" button)
   - [x] clears all settings other than hardcore and game settings and resets them back to otherwise initial conditions
- [x] should not tick when dead
- [x] can not join guilds [hidden in ui, also] or parties
- [x] should have all pets ascend for free w/ button click (text will need to be updated for hardcore players)
- [x] can not PvP
- [ ] can not salvage
   - [x] salvage API calls should fail and return failure message
- [x] is not affected by festivals
- [x] should not be able to get injuries (witch event)
- [x] should only die if they lose combat
- [x] should send a different message when joining the game
- [ ] should send a message when dying and leaving forever
- [ ] should track statistics on the main account. unsure of mechanism to link yet.
   - [ ] highest hardcore level
   - [ ] most ticks
   - [ ] most hardcore deaths
- [ ] should earn new achievements for the main account  
- [ ] cause of death should be tracked in statistics  

some class abilities will need tweaking:

- [x] barbarian (buff self physical stats, maybe)
- [ ] cleric (buff self mental stats, maybe)
- [x] bard (convert into a self buff)

stretch goal, maybe post-hardcore (@zig):

dead HC characters leave a temporary gravestone (probably rate-limited somehow in case of rapid deaths) that act as a boss encounter against the ghost of the HC character (with their stats/gear upon death) challengable by all players. Could be weird if a HC character dies on a boss, though.

**Comment by boozledorf** [comment]
I disagree with no achievements/collectibles/statistics, because that precludes pets.  I believe hardcore mode should be "if you die, you die" but other than that, it should be mostly the same game.  Player parties and guilds are understandable to exclude because that leaves your fate in the hands of someone else (guild leader, party barbarian, e.g.) but I don't see the benefit of losing out on stat bonuses from achievements.

**Comment by seiyria** [comment]
I still think premium/quests should be out, for the most part. But that makes sense.

**Comment by boozledorf** [comment]
Titles for the main player:
On hardcore character link: "the Hardcore"
On hardcore character deaths: "The Dead" (1), "The In Denial" (5), "The Persistent" (10), "The Zombie" (100)

**Comment by boozledorf** [comment]
In order to enable Hardcore Mode when creating a character, a secret key must be generated in an existing character's Settings page (this feature is disabled for Hardcore characters).  That key will be entered into the character creation page, and upon validation, will enable the Hardcore checkbox.  This key will then link the new Hardcore character to an already existing character for achievement purposes and bragging rights.
